src/components/ui/PieChartWithLegend.tsx

```diff
@@ -0,0 +1,86 @@
+import {merge} from 'lodash'
+import React from 'react'
+import {ChartComponentProps, Pie} from 'react-chartjs-2'
+
+import * as styles from './PieChartWithLegend.module.css'
+
+type DataPoint = {
+	value: number,
+	label: string,
+	backgroundColor?: string,
+	additional?: React.ReactNode[],
+}
+
+type Props = ChartComponentProps & {
+	headers?: {
+		label?: React.ReactNode,
+		additional?: React.ReactNode[],
+	},
+	data: DataPoint[],
+}
```

Comment:
`interface`s are nicer if you don't need to have mapped types or unions. They show up nicer on tooltips and can be extended without having to modify the original type definition.

```ts
interface DataPoint {
  value: number;
  ...
}

interface Props extends ChartComponentProps {
  headers?: ...
}
```